spring-data-jpa/src/main/antlr4/org/springframework/data/jpa/repository/query/Hql.g4

```diff
@@ -700,7 +700,7 @@ reservedWord
 	| FOR
 	| FORMAT
 	| FROM
-//	| FULL
+	| FULL
```

Comment:
What about the JPQL grammar about `FULL`?

Reply:
I've copied the test cases I used to expose this issue in HQL into the JPQL-based tests. Looks like "RIGHT" is okay, but we need to add support for "LEFT", "INNER", and "OUTER".